packages/core/src/fhir-deduplication/__tests__/deduplicate-fhir.test.ts

```diff
@@ -342,4 +353,41 @@ describe("deduplicateFhir", () => {
     expect(resDiagReport?.resourceType).toBe("DiagnosticReport");
     expect(resDiagReport?.result).toEqual([{ reference: observation2Ref }]);
   });
+
+  it("groups identical conditions referenced by an encounter", () => {
```

Comment:
Since the Conditions have diff IDs, I'd expect them not to be "grouped", based on the implementation of `removeDuplicateReferences`. What am I missing?

Reply:
the test was wrongly named. Name should be 
"removes duplicate diagnoses references from an Encounter"

Reply:
I was not referring to the name, but the implementation of the test.

We're sending two conditions with diff IDs and we're returning only one, while the implementation of `removeDuplicateReferences` only group them if they have the same ID.

What am I missing here? Or is the test/code incorrect somehow? Or maybe the test is not passing?

Also, we could test the function in isolation, that makes it easier to test diff scenarios, like I commented [here](https://github.com/metriport/metriport/pull/2711#discussion_r1744166488).

Reply:
The two conditions are identical. So first they are deduplicated, and one condition is merged into the other. That condition that is merged, all references to it across the bundle get replaced, and so the end result is an Encounter that originally referenced two different conditions, but now references the same condition twice. So then we deduplicate the references, so it only references the condition once. 

I can test the function individually too, though I think this test is good.